docs/csharp/misc/cs0825.md

```diff
@@ -10,7 +10,17 @@ ms.assetid: 49393d23-ec5f-4b44-a3fd-7e0a95ac0edd
 # Compiler Error CS0825
 The contextual keyword 'var' may only appear within a local variable declaration.  
   
- Implicit typing with the `var` keyword can only be applied to variables at local method scope.  
+ Implicit typing with the `var` keyword can only be applied to variables at local method scope. Implicit typing is not available for class fields as the C# compiler would encounter a logical paradox as it processed the code: the compiler needs to know the type of the field, but it cannot determine the type until the assignment expression is analyzed, and the expression cannot be evaluated without knowing the type. Consider the following code:
+
+ ```csharp
+private var bookTitles;
+ ```
+`bookTitles` is a class field given the type `var`. As the field has no expression to evaluate, it is impossible for the compiler to infer what type `bookTitles` is supposed to be. In addition, adding an expression to the field (like you would for a local variable) is also insufficient:
+
+```csharp
+private var bookTitles = new List<string>();
+```
+When the compiler encounters fields during code compilation, it records each field's type before processing any expressions associated with it. The compiler encounters the same paradox trying to parse `bookTitles`: it needs to know the type of the field, but the compiler would normally determine `var`'s type by evaluating the expression, which isn't possible without knowing the type beforehand. 
```

Comment:
Expressions are not evaluated by a compiler (except the constant ones). Above you use the "analyze" word in this context.

```suggestion
When the compiler encounters fields during code compilation, it records each field's type before processing any expressions associated with it. The compiler encounters the same paradox trying to parse `bookTitles`: it needs to know the type of the field, but the compiler would normally determine `var`'s type by analyzing the expression, which isn't possible without knowing the type beforehand. 
```

However, then the last part of the sentence becomes wrong.

@jmarkman I think you caught everything in this sentence:
>When the compiler encounters fields during code compilation, it records each field's type before processing any expressions associated with it.

Then logical paradox is clear:
 1. To infer type, the compiler needs to analyze the initialization expression.
 2. The compiler requires the field types to be known before checking any initialization expression.